hack/make-rules/test-cmd-util.sh

```diff
@@ -1613,7 +1613,7 @@ run_secrets_test() {
 
   ### Create a docker-registry secret in a specific namespace
   # Pre-condition: no SECRET exists
-  kube::test::get_object_assert 'secrets --namespace=test-secrets' "{{range.items}}{{$id_field}}:{{end}}" ''
+  kube::test::wait_object_assert 'secrets --namespace=test-secrets' "{{range.items}}{{$id_field}}:{{end}}" ''
```

Comment:
agree, which is why I don't think we should modify the test suite written for the core API server to accommodate behavior differences in federation like this... if the core API server stopped deleting secrets immediately, I'd consider that a bug and would want to know. if federation requires tests for different behavior, I think it should have separate tests